Code/src/main/java/nl/utwente/group10/ui/components/blocks/function/ArgumentSpace.java

```diff
@@ -0,0 +1,373 @@
+ * to control the amount of inputs that are to be applied (amount of currying).
+ * 
+ * The input arguments are ordered using the layoutX, and the knotIndex is based
+ * on this and the knot's layoutX. Usage of translateX is strictly for visual
+ * improvements.
+ * 
+ * Supports multi-touch interaction.
+ */
+public class ArgumentSpace extends Pane implements ComponentLoader {    
+    /** Horizontal space to keep between elements in ArgumentSpace. */
+    public static final double H_GAP = 10;
+    
+    /** Vertical height of the ArgumentSpace. */
+    public static final double HEIGHT = 50;
+    
+    /**
+     * Modifies the snap range of the knot. Lower value results in a wider snap
+     * range, effect is limited between -0.5 and 0.5.
+     */
+    public static final double KNOT_SNAP_MODIFIER = 0.3;
+    
+    /** The block to which this ArgumentSpace belongs. */
+    private FunctionBlock block;
+    
+    /** The knot, used to control the knot index. */
+    @FXML private Circle knot;
+    
+    /**
+     * The index of the knot, all inputs with index higher or equal to the
+     * knot are inactive, creating a function as output.
+     */
+    private IntegerProperty knotIndex;
+    
+    /** List of input arguments. */
+    private List<InputArgument> leftArguments;
+    
+    /** The argument containing the function's output, its return value. */
+    @FXML private Label rightArgument;
+    
+    /** 
+     * InputID of the current action (INPUT_ID_NONE if idle)
+     * The inputID is associated with an input (INPUT_ID_MOUSE for mouse, > 0 for touch).
+     * This is required to support multi-touch inputs.
+     */
+    private int inputID;
+    
+    /**
+     * Constructs an ArgumentSpace belonging to the function block as given.
+     * @param block FunctionBlock to which this ArgumentSpace belongs.
+     */
+    public ArgumentSpace(FunctionBlock block, int inputCount) {
+        this.loadFXML("ArgumentSpace");
+        
+        this.block = block;
+        this.inputID = ConnectionCreationManager.INPUT_ID_NONE;
+        leftArguments = new ArrayList<InputArgument>();
+        knotIndex = new SimpleIntegerProperty(0);
+        
+        knotIndex.addListener(event -> snapToKnotIndex());    
+        
+        //Create and attach InputArguments for the inputs.
+        for (int i = 0; i < inputCount; i++) {
+            InputArgument arg = new InputArgument(block);
+            // Make sure that when input widht changes, the total width also changes.
+            arg.getInputLabel().widthProperty().addListener(a -> Platform.runLater(block::updateLayout));
+            leftArguments.add(arg);
+            
+            if (i > 0) {
+                Region prev = leftArguments.get(i-1);
+                // The i-th (left)argument is placed to the right of argument i-1, with a horizontal space of H_GAP between them.
+                arg.layoutXProperty().bind(prev.layoutXProperty().add(prev.widthProperty()).add(H_GAP));
```

Comment:
I don't really like binding layoutX. It seems a bit magic, and like it could result in wasted cycles (recalculations causing other recalculations causing other recalculations...). See `ui-hbox` branch for an alternative if you haven't already.